msal-python-conceptual/advanced/migrate-python-adal-msal.md

```diff
@@ -0,0 +1,107 @@
+
+Supports:
+
+- Work and school accounts (Azure AD provisioned accounts)
+- Personal accounts (such as Outlook.com or Hotmail.com)
+- Your customers who bring their own email or social identity (such as LinkedIn, Facebook, Google) via the Azure AD B2C offering
+
+- Is standards compatible with:
+  - OAuth v2.0
+  - OpenID Connect (OIDC)
+
+For more information about MSAL, see [MSAL overview](../index.md).
+
+### Scopes not resources
+
+ADAL Python acquires tokens for resources, but MSAL Python acquires tokens for scopes. The API surface in MSAL Python doesn't have resource parameter anymore. You would need to provide scopes as a list of strings that declare the desired permissions and resources that are requested. To see some example of scopes, see [Microsoft Graph's scopes](/graph/permissions-reference).
+
+You can add the `/.default` scope suffix to the resource to help migrate your apps from the v1.0 endpoint (ADAL) to the Microsoft identity platform (MSAL). For example, for the resource value of `https://graph.microsoft.com`, the equivalent scope value is `https://graph.microsoft.com/.default`. If the resource isn't in the URL form, but a resource ID of the form `XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX`, you can still use the scope value as `XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXX/.default`.
+
+For more details about the different types of scopes, refer to [Permissions and consent in the Microsoft identity platform](/azure/active-directory/develop/permissions-consent-overview) and the [Scopes for a Web API accepting v1.0 tokens](/azure/active-directory/develop/msal-v1-app-scopes) articles.
+
+### Error handling
+
+ADAL for Python uses the exception `AdalError` to indicate that there's been a problem. MSAL for Python typically uses error codes, instead. For more information, see [MSAL for Python error handling](msal-error-handling-python.md).
+
+### API changes
+
+The following table lists an API in ADAL for Python, and the one to use in its place in MSAL for Python:
+
+| ADAL for Python API | MSAL for Python API |
+| ------------------- | ------------------- |
+| [AuthenticationContext](https://adal-python.readthedocs.io/en/latest/#adal.AuthenticationContext)                                                                                                                                                                                                                       | [PublicClientApplication](/python/api/msal/msal.application.publicclientapplication) or [ConfidentialClientApplication](/python/api/msal/msal.application.confidentialclientapplication)                                                                                                                                                                                                                                    |
```

Comment:
Given that all of these are now in the Python API browser, should we use the `xref` syntax instead to directly point to the items rather than have hardcoded relative links?

Reply:
I've always used the relative links out of habit for the \develop doc sets, and I understand that xref is also in the MSAL.NET docs. I see an example in use [here](https://learn.microsoft.com/en-us/entra/msal/dotnet/advanced/backup-authentication-system). Happy to switch if that's best practice!